pulsar/consumer_impl.go

```diff
@@ -580,12 +583,12 @@ func (c *consumer) ReconsumeLaterWithCustomProperties(msg Message, customPropert
 }
 
 func (c *consumer) Nack(msg Message) {
+	if !checkMessageIDType(msg.ID()) {
+		c.log.Warnf("invalid message id type %T", msg.ID())
+		return
+	}
 	if c.options.EnableDefaultNackBackoffPolicy || c.options.NackBackoffPolicy != nil {
 		mid := c.messageID(msg.ID())
-		if mid == nil {
```

Comment:
Even if we check the type of msgID, we can't remove this check, because there is other logic that returns `nil`

https://github.com/apache/pulsar-client-go/blob/48b7d0195327a81f638a3c51df2010d7eb244bbe/pulsar/consumer_impl.go#L755-L759

We can add `checkMessageIdType` to `c. messageID` method, and keep this check.

Reply:
Great suggestion! I will fix it.